charts/kubernetes-agent/templates/pod-rolebinding.yaml

```diff
@@ -17,4 +17,20 @@ roleRef:
   kind: ClusterRole
   name: {{ $podClusterRoleName }}
   apiGroup: rbac.authorization.k8s.io
+{{- end -}}
+{{- if and .Values.persistence.nfs.watchdog.enabled .Values.persistence.nfs.enabled}}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: {{ printf "%s-binding" $podServiceAccountFullName }}
+  namespace: {{ .Release.Namespace | quote }}
+subjects:
+  - kind: ServiceAccount
+    name: {{ $podServiceAccountName }}
+    namespace: {{ $.Release.Namespace | quote }}
+roleRef:
+  kind: ClusterRole
+  name: {{ $podClusterRoleName }}
+  apiGroup: rbac.authorization.k8s.io
```

Comment:
💭 Should we have a Role that is _just_ for the script pod that has only pod delete permissions 🤔
Is there a way we can stop if from being able to kill pods that aren't script pods?

Reply:
We definitely could have a specific role with only delete - we will need it to be at least namespace scoped, as that's the smallest scope 🙁

Reply:
Yeah, let's do that. Then we don't give the script pod itself too many permissions to the namespace of the agent

Reply:
Done in [c59228f](https://github.com/OctopusDeploy/helm-charts/pull/110/commits/c59228f19c8575e81f442e93c4b0e3ab8062dc80) :)